File: scripts/eoolo.java
public class eoolo {
    public static void main(String[] args){
        // Logical operators in java accept only boolean.
        System.out.println(expr1() || expr2());
    }

    private static boolean expr1(){
        System.out.println("method expr1");
        return true;
    }

    private static boolean expr2(){
        System.out.println("method expr2");
        return true;
    }
}

File: scripts/eoolo.f
      PROGRAM hello
          logical expr1, expr2
          WRITE (*,*) (expr1().or.expr2())
      END

      logical FUNCTION expr1() result(r)
          WRITE (*,*) 'method expr1'
          r = .TRUE.
      END

      logical FUNCTION expr2() result(r)
          WRITE (*,*) 'method expr2'
          r = .TRUE.
      END
